draft-ietf-tls-svcb-ech.md

```diff
@@ -90,6 +90,84 @@ Origins that publish an "ech" SvcParam in their HTTPS record SHOULD also publish
 
 If all HTTPS records for an alt-authority contain "ech" SvcParams, the client MUST adopt SVCB-reliant behavior (as in {{disabling-fallback}}) for that RRSet.  This precludes the use of certain connections that Alt-Svc would otherwise allow, as discussed in {{Section 9.3 of !SVCB}}.
 
+# Examples
+
+~~~
+$ORIGIN simple.example. ; Simple example zone
+@ 300 IN A     192.0.2.1
+         AAAA  2001:db8::1
+         HTTPS 1 . ech=ABC...
+www 300 IN A 192.0.2.1
+           AAAA 2001:db8::1
+           HTTPS 1 . ech=ABC...
+~~~
+{: title="Simple example zone with the same configuration on the apex and web domain."}
+
+~~~
+$ORIGIN cdn.example. ; CDN operator zone
+pool 300 IN A 192.0.2.1
+            AAAA 2001:db8::1
+            HTTPS 1 . ech=ABC...
+
+$ORIGIN customer.example. ; CDN customer's zone
+@   3600 IN HTTPS 0 pool.cdn.example.
+; Apex IP records for compatibility with clients that do not support
+; HTTPS records.
+@   300  IN A    192.0.2.1
+            AAAA 2001:db8::1
+
+www 300  IN CNAME pool.cdn.example.
+~~~
+{: title="ECH usage pattern for an aliasing-based CDN."}
+
+~~~
+$ORIGIN secret.example. ; High confidentiality zone
+www     3600 IN HTTPS 1 backend ech=ABC... mandatory=ech
+backend 300  IN A     192.0.2.1
+                AAAA  2001:db8::1
+~~~
+{: title="A domain that is only reachable using ECH."}
+
+~~~
+$ORIGIN cdn1.example. ; First CDN operator zone
+pool 300 IN A 192.0.2.1
+            AAAA 2001:db8::1
+            HTTPS 1 . ech=ABC...
+
+$ORIGIN cdn2.example. ; Second CDN operator zone
+pool 300 IN A 192.0.2.2
+            AAAA 2001:db8::2
+            HTTPS 1 . ech=DEF...
+
+;; Multi-CDN customer zone
+$ORIGIN customer.example. ; Multi-CDN customer's zone
+; In this configuration, the customer is combining HTTPS
+; records from the CDNs' zones, with appropriate modifications.
+; Automation is required to keep these records consistent with
+; the original records in the CDN providers' zones.
+@   3600 IN HTTPS 1 pool.cdn1.example. ech=ABC...
+            HTTPS 1 pool.cdn2.example. ech=DEF...
+; Apex IP records for compatibility with clients that do not support
+; HTTPS records.
+@   300  IN A    192.0.2.1
+            A    192.0.2.2
+            AAAA 2001:db8::1
+            AAAA 2001:db8::2
+
+www 3600  IN CNAME @
```

Comment:
What does a CNAME to @ mean here?  Following what we did in rfc9460:

```suggestion
; some times containing
www 3600  IN CNAME pool.cdn1.example.
; other times containing
www 3600  IN CNAME pool.cdn2.example.
```

Reply:
CNAME to @ reuses the apex records, which saves 5 lines in this example.  Maybe I'm not understanding the question.

Note that this example is showing "merged" multi-CDN, as opposed to "selected" multi-CDN, which is what we showed in RFC 9460.

Reply:
I think it's much better to show a "selected" multi-CDN example.
(As a CDN vendor, I don't see it being operationally viable for CDN vendors to offer a merged multi-CDN implementation.  The automation required for that is out-of-scope from this draft and would be a huge operational pain.)
Trying to show the zone apex case may be adding more confusion than it helps with.

Reply:
OK, I've incorporated these suggestions, with some additional tweaks.